python/lsst/meas/algorithms/detection.py

```diff
@@ -166,154 +234,154 @@ def setDefaults(self):
-To investigate the @ref meas_algorithms_detection_Debug, put something like
-@code{.py}
-    import lsstDebug
-    def DebugInfo(name):
-        di = lsstDebug.getInfo(name)        # N.b. lsstDebug.Info(name) would call us recursively
-        if name == "lsst.meas.algorithms.detection":
-            di.display = 1
-
-        return di
-
-    lsstDebug.Info = DebugInfo
-@endcode
-into your debug.py file and run measAlgTasks.py with the @c --debug flag.
     """
+
     ConfigClass = SourceDetectionConfig
     _DefaultName = "sourceDetection"
 
     def __init__(self, schema=None, **kwds):
-        """!Create the detection task.  Most arguments are simply passed onto pipe.base.Task.
+         """Create the detection task.  Most arguments are simply passed onto `pipe.base.Task`.
+         Parameters
+         ----------
+         schema: `lsst::afw::table::Schema`
+             used to create the output `lsst.afw.table.SourceCatalog`
+
+         kwds: 'list'
+             Keyword arguments passed to `lsst.pipe.base.task.Task.__init__.`
+
+         Notes
+         -----
+         This task can add fields to the schema, so any code calling this task must ensure that
+             these columns are indeed present in the input match list.
+             If schema is not None and configured for 'both' detections,
+             a 'flags.negative' field will be added to label detections made with a
+             negative threshold."""
+         pipeBase.Task.__init__(self, **kwds)
+         if schema is not None and self.config.thresholdPolarity == "both":
+             self.negativeFlagKey = schema.addField(
+                 "flags_negative", type="Flag",
+                 doc="set if source was detected as significantly negative"
+             )
+         else:
+             if self.config.thresholdPolarity == "both":
+                 self.log.warn("Detection polarity set to 'both', but no flag will be "
+                               "set to distinguish between positive and negative detections")
+             self.negativeFlagKey = None
+         if self.config.reEstimateBackground:
+             self.makeSubtask("background")
+         if self.config.doTempLocalBackground:
+             self.makeSubtask("tempLocalBackground")
+         if self.config.doTempWideBackground:
+             self.makeSubtask("tempWideBackground")
 
-        @param schema An lsst::afw::table::Schema used to create the output lsst.afw.table.SourceCatalog
-        @param **kwds Keyword arguments passed to lsst.pipe.base.task.Task.__init__.
+    @pipeBase.timeMethod
+    def run(self, table, exposure, doSmooth=True, sigma=None, clearMask=True, expId=None):
+        """Run source detection and create a SourceCatalog.
 
-        If schema is not None and configured for 'both' detections,
-        a 'flags.negative' field will be added to label detections made with a
-        negative threshold.
+        Parameters
+        ----------
+        table: 'lsst.afw.table.SourceTable'
```

Comment:
A few formatting issues with parameters like this. It Should look like:

```
table : `lsst.afw.table.SourceTable`
    Object that will be used to create the SourceCatalog.
```

Note the spaces around the colon and the use of backticks, not single quotes.

https://developer.lsst.io/python/numpydoc.html#py-docstring-parameters